lib/optimizely.rb

```diff
@@ -507,6 +513,14 @@ def is_valid
       config.is_a?(Optimizely::ProjectConfig)
     end
 
+    def close
+      return if @stopped
+
+      @stopped = true
+      @config_manager.stop! if @config_manager.respond_to?(:stop!)
```

Comment:
how you are making sure, that stop property exists for both `event_processor` and `config_manager`

Reply:
`respond_to?(:stop!) ` ensures that the instance method `stop` exists!

Reply:
I suggest `stop` should be treated as required for any `event_processor` or `config_manager` passed in. We could validate them in the constructor. When `stop` doesn't exist, we can log a message and ignore it.